### PR TITLE
Add test mode to interface

### PR DESCRIPTION
**Pull request by jherico**
Add a test mode to interface, triggered by using the `--testScript` command line option, followed by a string containing a local script to run.  An example script which automatically exits the application after 10 seconds is included.

This PR only adds the command line switch and the automated script running.  It does not add a reporting mechanism yet.

## Testing

Download [this script](https://raw.githubusercontent.com/jherico/hifi/test_mode/scripts/developer/tests/testTestMode.js) to your local system and run interface.exe from a command line with the option `--testScript <pathToDownloadedFile>`.  10 seconds after starting up, the application should automatically shut down.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2a43f5fbef5baba08ce3a7acb7ed92ffe722f78e available at: https://gist.github.com/cf624456ce3ba7e00642e4a5eb06867a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9190-2a43f5f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9190-2a43f5f.dmg

**Comment by OmegaHeron** [comment]
Windows 10
QA Pass

General smoke test *not* using command line startup, normal run mode.  Seems fine.

Using command line exec as specified (and for other duration vals)  Pass

Calling with an invalid script result in a "normal" interface run

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -1329,90 +1344,96 @@ Application::Application(int& argc, char** argv, QElapsedTimer& startupTimer, bo
...
+            QString line = contentVersionFile.readAll();
+            // toInt() returns 0 if the conversion fails, so we don't need to specifically check for failure
+            contentVersion = line.toInt();
+        }
+        qCDebug(interfaceapp) << "Server content version: " << contentVersion;
 
+        bool hasTutorialContent = contentVersion >= 1;
 
-    // Get sandbox content set version, if available
-    auto acDirPath = PathUtils::getRootDataDirectory() + BuildInfo::MODIFIED_ORGANIZATION + "/assignment-client/";
```
Comment (samcake) [inline interface/src/Application.cpp]: THese lifew next lines seem to be gone in your version.
Why that ?
I don;t understand what they do?

Comment (samcake) [inline interface/src/Application.cpp]: OH my bad, i seem them.
nevermind

**Pull request merged**